File: typescript-learning/mario.ts
import * as readlineSync from "readline-sync";

function getInt(promptText: string): number {
  return readlineSync.questionInt(promptText);
}

let height: number;

while (true) {
  height = getInt("Height: ");

  // Checking user input height falls within bounds
  if (height > 0 && height < 9) {
    break;
  }
}

for (let row = 1; row <= height; row++) {
  // Initialise string for the row
  let rowString = "";

  // Calculating number of spaces from left to the first brick
  const spaces = height - row;

  // Adding spaces between far left and the first brick
  for (let spaceCount = 0; spaceCount < spaces; spaceCount++) {
    rowString += " ";
  }

  // Adding left wall bricks
  for (let leftRow = 0; leftRow < row; leftRow++) {
    rowString += "#";
  }

  // Adding two spaces between left and right wall
  rowString += "  ";

  // Adding right wall bricks
  for (let rightRow = 0; rightRow < row; rightRow++) {
    rowString += "#";
  }

  // Printing row + adding line break for next row
  console.log(rowString);
}

File: typescript-learning/mario.py
def get_int(prompt_text):
    while True:
        try:
            return int(input(prompt_text))
        except ValueError:
            # Re-loops without crashing if the input isn't an integer
            pass

while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

for row in range(1, height + 1):
    spaces = height - row

    # Print amount of spaces on the left hand side for row
    for space_count in range(spaces):
        print(' ', end='')

    # Print left row of bricks
    for left_row in range(row):
        print('#', end='')

    # Print 2 spaces in between walls
    print('  ', end='')

    # Print right row of bricks
    for right_row in range(row):
        print('#', end='')

    # Add line break for next row
    print()